packages/marshal/src/types.js

```diff
@@ -2,70 +2,317 @@
+ * @typedef {*} Remote
+ * A `Remote<T>` is a Remotable object in one vat that represents a
+ * PrimaryRemotable of type T in another vat. The Remote has an unforgeable
+ * identity that is one-to-one with the unforgeable identity of its primary.
+ * If there is only one comm system between the two vats, then the identity of
+ * the Remote is not observably different than the identity of its primary.
  *
- * A Passable has a pass-by-copy superstructure. This includes the atomic
- * pass-by-copy primitives ("bigint" | "boolean" | "null" | "number" |
- * "string" | "undefined") and the composite pass-by-copy objects ("copyArray" |
- * "copyRecord" | "copyError"). The composite pass-by-copy objects that may
- * contain other Passables.
+ * A Remote, being a kind of Remotable, is pass-by-remote. When a Remote<T>
+ * is passed as an argument in a message to an object in another vat, it
+ * arrives as either a T or a Remote<T>, depending on whether the destination
+ * is the home vat of its primary.
+ */
+
+/**
+ * @template {PrimaryRemotable} T
+ * @typedef {T | Remote<T>} Remotable
+ * Corresponding PassStyle is "remotable"
  *
- * A Passable's pass-by-copy superstructure ends in PassableCap leaves. The
- * Passable can be further classified by the nature of these leaves. Since a
- * Passable is hardened, its structure and classification is stable --- its
- * structure and classification cannot change even if some of the objects are
- * proxies.
+ * A Remotable<T> is either the PrimaryRemotable T in its home vat, or is one
+ * of its Remote<T> representatives in other vats. The primary together with
+ * all of its remotes are, in aggregate, the Unum<T>, the abstact distributed
+ * object consisting of all these individual remotables. If there is only one
+ * coherent comm system among the vats in question, then the unum as a whole
+ * effectively has the unforgeable identity of its primary, since the identity
+ * of each of its remotables is not observably different than the identity of
+ * its primary.
+ *
+ * A Remotable is a PassableKey, which `sameKey` compares for euivalence
+ * according to its unforgeable identity, which effectively compares the
+ * unforgeable identities of their primaries.
+ *
+ * Remotables are unordered and cannot be sorted.
+ *
+ * A message eventually sent to a Remotable<T> is eventually delivered to
+ * its primary T in the home vat of the primary.
+ *
+ * A remotable is pass-by-remote. When a (primary or remote) remotable is
+ * passed as an argument in a message to an object in another vat, it arrives
+ * as a (primary or remote) remotable for the same primary.
+ *
+ * A Remotable can be a key in a WeakStore, i.e., it can be a key in
+ * a WeakMemoryMap or a WeakVirtualMap, or an element of a WeakMemorySet or
+ * a WeakVirtualSet. Note that there are no passable weak stores, since it
+ * doesn't make sense to pass something that cannot be enumerated.
+ */
+
+/**
+ * @typedef {Remotable | Promise} LeafCap
```

Comment:
This doesn't work as a definition (unspecified type parameters are invalid, so they degrade to just `any`).  You probably need to use:
```suggestion
 * @template T
 * @typedef {Remotable<T> | Promise<T>} LeafCap
```